presto-orc/src/test/java/com/facebook/presto/orc/TestStructStreamReader.java

```diff
@@ -0,0 +1,293 @@
+ * limitations under the License.
+ */
+
+package com.facebook.presto.orc;
+
+import com.facebook.presto.spi.ConnectorSession;
+import com.facebook.presto.spi.Page;
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.block.BlockBuilder;
+import com.facebook.presto.spi.block.RowBlock;
+import com.facebook.presto.spi.type.NamedTypeSignature;
+import com.facebook.presto.spi.type.RowFieldName;
+import com.facebook.presto.spi.type.StandardTypes;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.spi.type.TypeManager;
+import com.facebook.presto.spi.type.TypeSignatureParameter;
+import com.facebook.presto.testing.TestingConnectorSession;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import io.airlift.slice.Slices;
+import io.airlift.units.DataSize;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static com.facebook.presto.memory.context.AggregatedMemoryContext.newSimpleAggregatedMemoryContext;
+import static com.facebook.presto.orc.OrcEncoding.ORC;
+import static com.facebook.presto.orc.OrcTester.HIVE_STORAGE_TIME_ZONE;
+import static com.facebook.presto.orc.TestingOrcPredicate.ORC_ROW_GROUP_SIZE;
+import static com.facebook.presto.orc.TestingOrcPredicate.ORC_STRIPE_SIZE;
+import static com.facebook.presto.orc.metadata.CompressionKind.NONE;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static io.airlift.units.DataSize.Unit.MEGABYTE;
+import static org.joda.time.DateTimeZone.UTC;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNull;
+
+@Test(singleThreaded = true)
+public class TestStructStreamReader
+{
+    private static final TypeManager TYPE_MANAGER = new TypeRegistry();
+
+    private static final Type TEST_DATA_TYPE = VARCHAR;
+
+    private static final String STRUCT_COL_NAME = "struct_col";
+
+    public static final ConnectorSession SESSION =
+            new TestingConnectorSession(com.google.common.collect.ImmutableList.of());
+
+    private TempFile tempFile;
+
+    @BeforeMethod
+    public void setUp()
+    {
+        tempFile = new TempFile();
+    }
+
+    /**
+     * Reader and writer have the same fields. Checks that fields are read in correctly
+     *
+     */
+    @Test
+    public void testValuesAreReadInCorrectly() throws IOException
+    {
+        List<String> readerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+        List<String> writerFields = new ArrayList<>(Arrays.asList("field_a", "field_b", "field_c"));
+        List<String> writerData = new ArrayList<>(Arrays.asList("field_a_value",
```

Comment:
no wrap.. basically don't rap unless it is really long 180, or there is something interesting at the end of the line